cmd/agent/agent.go

```diff
@@ -175,12 +180,12 @@ func run(c *cli.Context) error {
 		return err
 	}
 
-	agentID, err = client.RegisterAgent(ctx, platform, engine.Name(), version.String(), parallel)
+	agentConfig.AgentID, err = client.RegisterAgent(ctx, platform, engine.Name(), version.String(), parallel)
```

Comment:
I would introduce a new variable to avoid confusion and treat `agentConfig` as an immutable since it should represent the actual config.

Reply:
we save it later so we have to change agentConfig anyway